src/ddsrt/src/events/posix/events_posix.c

```diff
@@ -0,0 +1,301 @@
+/*
```

Comment:
I am stopping here: I suspect the preceding comments will apply quite naturally in the remainder and so the additional benefit of the additional effort is likely to be small.